memory_model/src/guest_memory.rs

```diff
@@ -113,13 +113,24 @@ impl GuestMemory {
 
     /// Returns true if the given address is within the memory range available to the guest.
     pub fn address_in_range(&self, addr: GuestAddress) -> bool {
-        addr < self.end_addr()
+        for region in self.regions.iter() {
+            if addr >= region.guest_base && addr < region_end(region) {
+                return true;
+            }
+        }
+        false
     }
 
     /// Returns the address plus the offset if it is in range.
     pub fn checked_offset(&self, addr: GuestAddress, offset: usize) -> Option<GuestAddress> {
-        addr.checked_add(offset)
-            .and_then(|a| if a < self.end_addr() { Some(a) } else { None })
+        if let Some(tgt) = addr.checked_add(offset) {
```

Comment:
nit: I think it would be nice to rename `tgt` to something more intuitive?

Reply:
renamed addr as base and tgt as addr, so it becomes addr = base + offset.